KPP/fullchem/fullchem.eqn

```diff
@@ -1516,16 +1510,16 @@ HOCl + HCl = Cl2 + H2O :       HOClUptkByHCl( State_Het );
 HOCl + HBr = BrCl + H2O :      HOClUptkByHBr( State_Het );                                                  {2014/02/03; Eastham2014; SDE}
 HOCl + SALACL = Cl2 + H2O :    HOClUptkBySALACL( State_Het );                                               {2018/01/22; XW}
 HOCl + SALCCL = Cl2 + H2O :    HOClUptkBySALCCL( State_Het );                                               {2018/01/22; XW}
-HOCl = SO4H3 + HCl :           HOClHetSO4H3( State_Het );                                                   {2018/11/08; XW}
-HOCl = SO4H4 + HCl :           HOClHetSO4H4( State_Het );                                                   {2018/11/08; XW}
+HOCl + HSO3m = SO4s + HCl :     HOClUptkByHSO3m( State_Het );                                                {2018/11/08; XW; June 6, 2021, MSL}
+HOCl + SO3mm = SO4s + HCl :     HOClUptkBySO3mm( State_Het );                                                {2018/11/08; XW; June 6, 2021, MSL}
 HOBr + HBr = Br2 + H2O :       HOBrUptkByHBr( State_Het );                                                  {2017/09/22; Sherwen2016b;TS,JAS,SDE}
 HOBr + HCl = BrCl + H2O :      HOBrUptkByHCl( State_Het );                                                  {2017/09/22; Sherwen2016b;TS,JAS,SDE}
 HOBr + SALACL = BrCl + H2O :   HOBrUptkBySALACL( State_Het );                                               {2018/01/22; Sherwen2017;TS,JAS,SDE;XW}
 HOBr + SALCCL = BrCl + H2O :   HOBrUptkBySALCCL( State_Het );                                               {2018/01/22; Sherwen2017;TS,JAS,SDE,XW}
 HOBr + BrSALA = Br2 :          HOBrUptkByBrSALA( State_Het );                                               {2017/09/22; Sherwen2017;TS,JAS,SDE}
 HOBr + BrSALC = Br2 :          HOBrUptkByBrSALC( State_Het );                                               {2017/09/22; Sherwen2017;TS,JAS,SDE}
-HOBr = SO4H1 + HBr :           HOBrHetSO4H1( State_Het );                                                   {2017/11/15; Chen2017; QJC}
-HOBr = SO4H2 + HBr :           HOBrHetSO4H2( State_Het );                                                   {2017/11/15; Chen2017; QJC}
+HOBr + HSO3m = SO4s + HBr :     HOBrUptkByHSO3m( State_Het );                                                {2017/11/15; Chen2017; QJC; June 6, 2021, MSL}
+HOBr + SO3mm = SO4s + HBr :     HOBrUptkBySO3mm( State_Het );                                                {2017/11/15; Chen2017; QJC; June 6, 2021, MSL}
```

Comment:
Should the product be SO4 to replicate the old code? Also for HOCl reactions.

Reply:
Not sure, that's what @msl3v (aka @michael-s-long) implemented.  I would let him reply.

Reply:
@viral211: For clarification
```
HOBr = SO4H1 + HBr :                         HET(ind_HOBr,7);                 
HOBr = SO4H2 + HBr :                         HET(ind_HOBr,8);   
```
were computed by this old code:
```F90
      !----------------------------------------------------------------------
      ! HOBr + HSO3-(aq) (update: XW 2019-06-08)
      !----------------------------------------------------------------------

      ! This reaction is first order, so no kII calculation is required
      kITemp = HETHOBr_TCld( XDenA,       rLiq,        rIce,                 &
                             ALiq,        AIce,        VAir,                 &
                             TempK,       CldFr,       hConc_Sul,            &
                             hConc_LCl,   hConc_ICl,   clConc_CldA,          &
                             clConc_CldC, clConc_Cldg, brConc_CldA,          &
                             brConc_CldC, brConc_Cldg, HSO3conc_Cld,         &
                             SO3conc_Cld, hno3_th,     hcl_th,               &
                             hbr_th,      7,           H                    )

      ! Make sure sulfate produced is less than SO2 available (qjc, 06/20/16)
      HET(ind_HOBr,  7) = kITemp * fupdateHOBr

      !----------------------------------------------------------------------
      ! HOBr + SO3--(aq) (update: XW 2019-06-08)
      !----------------------------------------------------------------------
      ! This reaction is first order, so no kII calculation is required
      kITemp = HETHOBr_TCld( XDenA,       rLiq,        rIce,                 &
                             ALiq,        AIce,        VAir,                 &
                             TempK,       CldFr,       hConc_Sul,            &
                             hConc_LCl,   hConc_ICl,   clConc_CldA,          &
                             clConc_CldC, clConc_Cldg, brConc_CldA,          &
                             brConc_CldC, brConc_Cldg, HSO3conc_Cld,         &
                             SO3conc_Cld, hno3_th,     hcl_th,               &
                             hbr_th,      8,           H                    )

      ! Make sure sulfate produced is less than SO2 available (qjc, 06/20/16)
      HET(ind_HOBr,  8) = kITemp * fupdateHOBr
```
and similarly for HOCl.

Reply:
Viral, I think you're correct. There was (is still, maybe) a drop in SO4s in the new version. And I somehow incorrectly interpreted that as being due to the removal of SO4H[1-4]. This should probably be S4, not SO4s. The question remains: Where did the SO4s go? All I can tell is that SO4s is created in a partitioningoperatoin within CHEM_SO4(). Any thoughts?

Reply:
Ah! It looks like SO4s is set from PSO4F from SEASALT_CHEM() and SEASALT_CHEM() isn't being run. I'll try to get a fix in.

Reply:
@viral211, @msl3v, maybe this is just the way it is with the KPP implementation.